File: library/swift/src/ResponseTrailersBuilder.swift
import Foundation

/// Builder used for constructing instances of `ResponseTrailers`.
@objcMembers
public final class ResponseTrailersBuilder: HeadersBuilder {
  /// Build the request trailers using the current builder.
  ///
  /// - returns: New instance of request trailers.
  public func build() -> ResponseTrailers {
    return ResponseTrailers(headers: self.headers)
  }
}

File: library/cc/response_trailers_builder.cc
#include "response_trailers_builder.h"

namespace Envoy {
namespace Platform {

ResponseTrailersSharedPtr ResponseTrailersBuilder::build() const {
  ResponseTrailers* trailers = new ResponseTrailers(this->allHeaders());
  return ResponseTrailersSharedPtr(trailers);
}

} // namespace Platform
} // namespace Envoy
